libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -359,9 +359,13 @@ where
         };
 
         self.connections.remove(endpoint);
-        self.life_cycle
-            .remove(meta.id)
-            .map_err(|err| ConnectionManagerError::ConnectionRemovalError(format!("{:?}", err)))?;
+        // remove mesh id, this may happen before reconnection is attempted
```

Comment:
Would it be better if mesh, instead of removing it completely when it is disconnected, moves it to another state 'disconnected' (maybe a self.disconnected_connections map), and then have this function look there as well, so that we can still detect programming errors?

Reply:
That worked. Removed this fix and updated mesh